spec/trigger_lock_spec.rb

```diff
@@ -76,5 +76,11 @@
       expect(trigger_lock_2.locked?).to be(true)
       expect(trigger_lock_1.unlock).to be(false)
     end
+
+    it "should not acquire the lock if there is an error" do
```

Comment:
`ConnectionPool#with` (`redis_pool`) is private so it feels odd to raise on that